### PR TITLE
DropMarket impl, ListMarket update

### LINKED ISSUE #246: [operator] listmartket RPC update
We should return:

a) Account Index of that market to allow deletion in #281 
b) Return current price, This should change the protobuf operator file (i.e marketInfo type and listMarket RPC)

### LINKED ISSUE #281: Add DropMarket RPC
Sometimes it coudl happen to create a new market by mistake (ie. calling DepositMarket RPC w/o arguments). This makes the crawler to watch that address for deposits and until it's not done, it will log warning in the logs whihc is annoying. 

There shoudl be a way to drop this market (if not funded the first time could be a nice check, altough no-one prevent us to drop already funded market anyway)

### PR DESCRIPTION
**Pull request by sekulicd**
This adds DropMarket RPC implementation and ListMarket update.

This closes #246 
This closes #281 

@tiero @altafan please review.

### CODE REVIEW

File: internal/infrastructure/storage/db/badger/market_repository_impl.go [hunk 1]
```diff
@@ -378,3 +378,23 @@ func restoreStrategy(market *domain.Market) {
...
+	var err error
+
+	if ctx.Value("tx") != nil {
+		tx := ctx.Value("tx").(*badger.Txn)
+		err = m.db.Store.TxDelete(tx, accountIndex, domain.Market{})
+	} else {
+		err = m.db.Store.Delete(accountIndex, domain.Market{})
+	}
+	if err != nil {
+		if err != badgerhold.ErrNotFound {
```
Comment (altafan) [inline internal/infrastructure/storage/db/badger/market_repository_impl.go:395]:
```suggestion
	if err != nil && err != badgerhold.ErrNotFound {
```

File: internal/infrastructure/storage/db/inmemory/market_memory_repository_impl.go [hunk 2]
```diff
@@ -181,6 +181,15 @@ func (r *MarketRepositoryImpl) UpdatePrices(_ context.Context, accountIndex int,
 	return nil
 }
 
+func (r *MarketRepositoryImpl) DeleteMarket(
+	_ context.Context,
+	accountIndex int,
+) error {
+	delete(r.db.marketStore.markets, accountIndex)
```
Comment (altafan) [inline internal/infrastructure/storage/db/inmemory/market_memory_repository_impl.go:188]: better to do this only if the market actually exists? like:
```
if _, ok := r.db.marketStore.markets[accountIndex]; ok {
  delete(r.db.marketStore.markets, accountIndex)
}
```

File: internal/interfaces/grpc/handler/operator.go [hunk 3]
```diff
@@ -726,20 +759,28 @@ func (o operatorHandler) listMarket(ctx context.Context, req *pb.ListMarketReque
 				return nil, err
 			}
 
-			pbMarketInfos := make([]*pb.MarketInfo, len(marketInfos), len(marketInfos))
+			pbMarketInfos := make([]*pb.MarketInfo, 0, len(marketInfos))
```
Comment (altafan) [inline internal/interfaces/grpc/handler/operator.go:762]: could you rever this? the existing way of filling the slice is actually less performant than this one. We should move to this solution everywhere else. I remember louis have already pointed this out previously.

File: internal/core/application/operator_service.go [hunk 4]
```diff
@@ -592,10 +593,11 @@ func (o *operatorService) ListMarket(
 		return nil, err
 	}
 
-	marketInfos := make([]MarketInfo, len(markets))
+	marketInfos := make([]MarketInfo, 0, len(markets))
```
Comment (altafan) [inline internal/core/application/operator_service.go:596]: and also here

File: internal/interfaces/grpc/handler/operator.go (hunk 3 above)
Comment (sekulicd) [reply internal/interfaces/grpc/handler/operator.go:762]: I am not sure that this is correct, i did benchmark and results are pretty much similar, and IMHO i think using append is more idiomatic.

````
func BenchmarkAppendOne(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendOne()
	}
}

func BenchmarkAppendTwo(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendTwo()
	}
}

func appendOne() {
	markets := make([]domain.Market, 0)
	for i := 0; i < 100; i++ {
		markets = append(markets, domain.Market{
			AccountIndex: 0,
			BaseAsset:    "",
			QuoteAsset:   "",
			Fee:          0,
			Tradable:     false,
			Strategy:     mm.MakingStrategy{},
			Price:        domain.Prices{},
		})
	}

	marketInfos := make([]MarketInfo, 0, len(markets))

	for _, market := range markets {
		marketInfos = append(marketInfos, MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		})
	}
}

func appendTwo() {
	markets := make([]domain.Market, 0)
	for i := 0; i < 100; i++ {
		markets = append(markets, domain.Market{
			AccountIndex: 0,
			BaseAsset:    "",
			QuoteAsset:   "",
			Fee:          0,
			Tradable:     false,
			Strategy:     mm.MakingStrategy{},
			Price:        domain.Prices{},
		})
	}

	marketInfos := make([]MarketInfo, len(markets))

	for index, market := range markets {
		marketInfos[index] = MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		}
	}
}
````

````
///////////////////////
BenchmarkAppendOne
BenchmarkAppendOne-8   	   66777	     20478 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   73107	     16236 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   65511	     16411 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   66888	     16502 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   58797	     17023 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	   60243	     18762 ns/op
PASS

////////////////////
BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   54678	     21754 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   59526	     17693 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   65001	     16569 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   63954	     18000 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   62292	     16478 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	   69199	     17819 ns/op
PASS
````

File: internal/infrastructure/storage/db/badger/market_repository_impl.go (hunk 1 above)
Comment (sekulicd) [reply internal/infrastructure/storage/db/badger/market_repository_impl.go]: Done

File: internal/interfaces/grpc/handler/operator.go (hunk 3 above)
Comment (altafan) [reply internal/interfaces/grpc/handler/operator.go:762]: did you take a look at this https://stackoverflow.com/a/38654841?

**Comment by tiero** [comment]
solve conflicts

**Comment by sekulicd** [comment]
> solve conflicts

Done

File: internal/interfaces/grpc/handler/operator.go (hunk 3 above)
Comment (sekulicd) [reply internal/interfaces/grpc/handler/operator.go:762]: Discussed offline with @altafan and improved testing, PSB:

```
var markets []domain.Market

func init() {
	markets := make([]domain.Market, 0)
	for i := 0; i < 100000; i++ {
		markets = append(markets, domain.Market{
			AccountIndex: 0,
			BaseAsset:    "",
			QuoteAsset:   "",
			Fee:          0,
			Tradable:     false,
			Strategy:     mm.MakingStrategy{},
			Price:        domain.Prices{},
		})
	}
}

func BenchmarkAppendOne(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendOne(markets)
	}
}

func BenchmarkAppendTwo(b *testing.B) {
	for i := 0; i < b.N; i++ {
		appendTwo(markets)
	}
}

func appendOne(markets []domain.Market) {

	marketInfos := make([]MarketInfo, 0, len(markets))

	for _, market := range markets {
		marketInfos = append(marketInfos, MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		})
	}
}

func appendTwo(markets []domain.Market) {
	marketInfos := make([]MarketInfo, len(markets), len(markets))

	for index, market := range markets {
		marketInfos[index] = MarketInfo{
			AccountIndex: uint64(market.AccountIndex),
			Market: Market{
				BaseAsset:  market.BaseAsset,
				QuoteAsset: market.QuoteAsset,
			},
			Fee: Fee{
				BasisPoint: market.Fee,
			},
			Tradable:     market.Tradable,
			StrategyType: market.Strategy.Type,
			Price:        market.Price,
		}
	}
}



BenchmarkAppendOne
BenchmarkAppendOne-8   	172570485	         6.78 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	168643347	         6.88 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	174481154	         6.80 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	175577139	         6.79 ns/op
PASS

BenchmarkAppendOne
BenchmarkAppendOne-8   	179474529	         6.83 ns/op
PASS

/////////////////////////////////////

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	174470482	         6.75 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	173583262	         6.75 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	172803805	         7.20 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	174627186	         6.71 ns/op
PASS

BenchmarkAppendTwo
BenchmarkAppendTwo-8   	174373942	         6.85 ns/op
PASS
```

@tiero what do you think?

**Pull request merged**